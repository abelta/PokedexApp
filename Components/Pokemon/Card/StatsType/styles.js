import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  typesStats: {
    flexDirection: 'row',
    padding: 2,
  },
});

export default styles;
