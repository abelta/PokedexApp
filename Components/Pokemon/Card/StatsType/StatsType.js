import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import { pokemonTypeShape } from '../../../../PropTypes/Pokemon';
import { BulletType } from '../.';
import styles from './styles';

const StatsType = ({ data, style }) => (
  <View style={[styles.typesStats, style]}>
    {
      data.map(type => <BulletType key={type.type.name} value={type.type.name} />)
    }
  </View>
);

StatsType.propTypes = {
  data: PropTypes.arrayOf(pokemonTypeShape).isRequired,
  style: PropTypes.number, // eslint-disable-line react/require-default-props
};

export default StatsType;
