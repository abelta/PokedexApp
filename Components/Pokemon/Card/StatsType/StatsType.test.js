import React from 'react';
import renderer from 'react-test-renderer';
import camelcaseKeys from 'camelcase-keys';
import StatsType from '.';
import data from './mock-data.json';

describe('StatsType', () => {
  it('renders properly', () => {
    const tree = renderer.create(
      <StatsType data={camelcaseKeys(data, { deep: true }).types} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
