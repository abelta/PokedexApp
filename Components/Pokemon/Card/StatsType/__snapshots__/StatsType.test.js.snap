// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsType renders properly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "padding": 2,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "margin": 2,
        },
        Object {
          "flex": 1,
        },
        Object {
          "backgroundColor": "#A890F0",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "flex": 1,
          "textAlign": "center",
        }
      }
    >
      flying
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "margin": 2,
        },
        Object {
          "flex": 1,
        },
        Object {
          "backgroundColor": "#F08030",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "flex": 1,
          "textAlign": "center",
        }
      }
    >
      fire
    </Text>
  </View>
</View>
`;
